R/bootstrap.R

```diff
@@ -117,39 +120,63 @@ is_bs_theme <- function(x) {
 #' @seealso [getCurrentOutputInfo()], [bootstrapLib()], [htmltools::tagFunction()]
 #' @examples
 #'
-#' library(bootstraplib)
-#' myThemableContent <- htmltools::tagFunction(function() {
-#'   theme <- getCurrentTheme()
-#'   if (is_bs_theme(theme)) {
-#'     tagList(
-#'       tags$head(
-#'         tags$style(HTML(
-#'           bs_sass(".text-50 { color: mix($body-bg, $body-color, 50%); }", theme)
-#'         ))
-#'       ),
-#'       h3(
-#'         "This color is 50% background and 50% foreground color",
-#'         class = "text-50"
-#'       )
-#'     )
-#'   } else {
-#'     "Try setting a theme = bs_theme()"
-#'   }
-#' })
-#'
 #' if (interactive()) {
+#'   library(bootstraplib)
 #'   ui <- fluidPage(
-#'     theme = bs_theme(bg = "yellow", fg = "red"),
-#'     myThemableContent
+#'     theme = bs_theme(),
+#'     h3("The current theme's primary color is:"),
+#'     uiOutput("primary"),
+#'     textInput("primary_input", label = "", placeholder = "Choose a new primary color")
 #'   )
-#'   shinyApp(ui, function(input, output) {})
+#'   server <- function(input, output, session) {
+#'     get_primary <- function() bs_get_variables(getCurrentTheme(), "primary")
+#'     primary <- reactiveVal(get_primary())
+#'     output$primary <- renderUI({
+#'       tags$span(primary(), style = paste("color: ", primary()))
+#'     })
+#'     # TODO: simplify & tryCatch
+#'     observeEvent(input$primary_input, {
+#'       new_theme <- bs_theme_update(getCurrentTheme(), primary = input$primary_input)
+#'       setCurrentTheme(new_theme)
+#'       primary(get_primary())
+#'     }, ignoreInit = TRUE)
+#'
+#'   }
+#'   shinyApp(ui, server)
 #' }
 #'
 #' @export
 getCurrentTheme <- function() {
+  # TODO: we still might need to elevate this to a reactive read
```

Comment:
One obstacle to making this reactive: app-level vs. session-level scoping. In most cases, when this is called, the session doesn't yet exist.